### PR TITLE
[WIP] Resequence

### PR DESCRIPTION
**Pull request by JelF**
Added `Verifly::Resequence` component to challenge callback hell

- [x] Updated to rubocop-config-umbrellio
- [x] Added basic resequence process
- [x] Get full spec and documentation coverage
- [x] Add modern callback system module into gem (service-based?)

Bonus:
- fixed bug in Applicator::MethodExtractor when method on biniding receiver was evaluated instead of
variable with same name

### CODE REVIEW

File: lib/verifly/dependent_callbacks/callback_group.rb [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+          -> { callback.call_around(binding_, *context, &reduced) }
+        end
+
+        reduced.call
+
+        result
+      end
+
+      def digest
+        [name, *list.map(&:hash)].hash
```
Comment (TrumpClone) [inline lib/verifly/dependent_callbacks/callback_group.rb]: `[name, *list].hash` m?

**Comment by JelF** [comment]
TODO: read https://habrahabr.ru/post/100953/

**Comment by JelF** [comment]
We should use Hash#tsort (stdlib) instead of re-implementing it

**Comment by JelF** [comment]
> - [ ] (_Optional_) add `Verifly::Resequence::Graph#to_dot` method which would allow to see callback hell in graphviz
> - [ ] update README.md (closes #6). @akxcv  review required there

IT probably could be done in separated PRs. After review ill increment version and merge it

**Review by TrumpClone — APPROVED** [review]

**Pull request merged**